src/soar_sdk/apis/artifact.py

```
@@ -0,0 +1,81 @@
+        """
+        Create a new artifact.
+
+        :param artifact: The artifact data to create.
+        """
+        artifact.update(
+            {k: v for k, v in self._artifact_common.items() if (not artifact.get(k))}
+        )
+        try:
+            json.dumps(artifact)
+        except TypeError as e:
+            error_msg = (
+                f"Artifact could not be converted to a JSON string. Error: {e!s}"
+            )
+            raise ActionFailure(error_msg) from e
+
+        if is_client_authenticated(self.soar_client.client):
+            client = self.soar_client.client
+            endpoint = "rest/artifact"
+            headers = {"Referer": f"{client.base_url}/{endpoint}"}
+            try:
+                response = client.post(endpoint, headers=headers, json=artifact)
+            except Exception as e:
+                error_msg = f"Failed to add artifact: {e}"
+                raise SoarAPIError(error_msg) from e
+
+            resp_data = response.json()
+
+            if "existing_artifact_id" in resp_data:
+                logger.info("Artifact already exists")
+                return resp_data["existing_artifact_id"]
+
+            if "id" in resp_data:
+                return resp_data["id"]
+
+            msg_cause = resp_data.get("message", "NONE_GIVEN")
+            message = f"Artifact addition failed, reason from server: {msg_cause}"
+            raise SoarAPIError(message)
+        else:
+            next_artifact_id = self.__add_artifact_locally(artifact)
+            return next_artifact_id
+
+    def __add_artifact_locally(self, artifact: dict) -> int:
+        if "container_id" not in artifact:
+            message = "Artifact addition failed, no container ID given"
+            raise SoarAPIError(message)
+
+        next_artifact_id = (max(self.__artifacts.keys()) if self.__artifacts else 0) + 1
+        self.__artifacts[next_artifact_id] = artifact
+        return next_artifact_id
```

Is the purpose of this to create a mock for when testing locally? If so, I think we should be more specific about testing whether we are on a SOAR server or not. 

If we're on a real SOAR instance and the client isn't authenticated, that's a problem -- we want to raise an exception about it, not just silently mock it.

---

It's just a helper function that adds an artifact when the soar client isn't authenticated. There's really no need for it, will 🪓